gear-programs/erc20-relay/app/src/services/erc20_relay.rs

```diff
@@ -0,0 +1,274 @@
+// Incorporate code generated based on the IDL file
+#[allow(dead_code)]
+mod vft {
+    include!(concat!(env!("OUT_DIR"), "/vft-gateway.rs"));
+}
+
+use super::*;
+use ops::ControlFlow::*;
+use sails_rs::{calls::ActionIo, gstd};
+use vft::vft_gateway;
+
+#[derive(Encode, TypeInfo)]
+#[codec(crate = sails_rs::scale_codec)]
+#[scale_info(crate = sails_rs::scale_info)]
+enum Event {
+    Relayed {
+        fungible_token: ActorId,
+        to: ActorId,
+        amount: U256,
+    },
+}
+
+pub struct Erc20Relay<'a, ExecContext> {
+    state: &'a RefCell<State>,
+    exec_context: ExecContext,
+}
+
+#[sails_rs::service(events = Event)]
+impl<'a, T> Erc20Relay<'a, T>
+where
+    T: ExecContext,
+{
+    pub fn new(state: &'a RefCell<State>, exec_context: T) -> Self {
+        Self {
+            state,
+            exec_context,
+        }
+    }
+
+    pub fn vft_gateway(&self) -> Option<ActorId> {
+        self.state.borrow().vft_gateway
+    }
+
+    pub fn set_vft_gateway(&mut self, vft_gateway: Option<ActorId>) {
+        let source = self.exec_context.actor_id();
+        let mut state = self.state.borrow_mut();
+        if source != state.admin {
+            panic!("Not admin");
+        }
+
+        state.vft_gateway = vft_gateway;
+    }
+
+    pub async fn relay(&mut self, message: EthToVaraEvent) -> Result<(), Error> {
+        let Some(vft_gateway_id) = self.state.borrow().vft_gateway else {
+            return Err(Error::AbsentVftGateway);
+        };
```

Comment:
Why not to set it in `init` then? to not deal with `Option`

Reply:
Yep, it is possible. However then we need to implement such logic in `vft-gateway` program. So I decided to make it here

Reply:
I mean that both programs must have each other's addresses when initialized. Unfortunately, this cannot be done without Option .

However, the question remains, where to place such logic: here in erc 20-relay-app or in vft-gateway?

Reply:
discussed: use Dfeault::default entity